mapbox/dr-ui
# Create `Edit` and `CodeSnippet` components

This PR adds a new component that you can send code and metadata to and these buttons will bring it into your favorite editor:

![image](https://user-images.githubusercontent.com/2180540/70103886-487f2e80-160a-11ea-8e24-41d11b1e1b4c.png)

This PR also adds CodeSnippet that include:
- option to add Edit in buttons
- option add CodeSnippetTitle
- automatically pulls in prism.css


To do:
* [x] more tests, especially to catch errors with prettier attempting to parse bad html
* [x] break code into smaller functions, add tests
* [x] rework props to simplify (html, css, js, resources, frontmatter) we are currently doing a lot of massaging and pieces together of data to get it how we want, instead we should provide helper functions to help extract data so that this component can stay simple


## How to test /mapbox-gl-js-docs:
1. from this branch, `npm run build` (build it locally)
2. in mapbox-gl-js-docs, pull down the `dr-ui-edit-buttons` branch
3. in mapbox-gl-js-docs, install dr-ui locally: `"@mapbox/dr-ui": "file:/Users/katydecorah/Documents/GitHub/dr-ui/pkg",` (update this with your path to your local dr-ui)
4. in mapbox-gl-js-docs, `yarn install` then `yarn start`
5. navigate to an examples page and test any of the edit buttons

## How to test in /help:

1. from this branch, `npm run build` (build it locally)
2. in help, pull down the `dr-ui-code-snippet` branch
3. in help, install dr-ui locally: `"@mapbox/dr-ui": "file:/Users/katydecorah/Documents/GitHub/dr-ui/pkg",` (update this with your path to your local dr-ui)
4. in help, `npm install` then `npm start`
5. navigate to http://localhost:8080/help/tutorials/use-mapbox-gl-js-with-react/  and see the CodeSnippet in action
